File: win-prog/ProgWinVB/Images and Bitmaps/CenterPixelSizeImage/CenterPixelSizeImage.vb
'-----------------------------------------------------
' CenterPixelSizeImage.vb (c) 2002 by Charles Petzold
'-----------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class CenterPixelSizeImage
    Inherits PrintableForm

    Private img As image

    Shared Shadows Sub Main()
        Application.Run(new CenterPixelSizeImage())
    End Sub

    Sub New()
        Text = "Center Pixel-Size Image"
        img = Image.FromFile("..\..\Apollo11FullColor.jpg")
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        grfx.DrawImage(img, (cx - img.Width) \ 2, _
                            (cy - img.Height) \ 2, _
                            img.Width, img.Height)
    End Sub
End Class

File: win-prog/ProgWinCS/Images and Bitmaps/CenterPixelSizeImage/CenterPixelSizeImage.cs
//---------------------------------------------------
// CenterPixelSizeImage.cs © 2001 by Charles Petzold
//---------------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class CenterPixelSizeImage: PrintableForm
{
	Image image;

	public new static void Main()
	{
		Application.Run(new CenterPixelSizeImage());
	}
	public CenterPixelSizeImage()
	{
		Text = "Center Pixel-Size Image";

		image = Image.FromFile("..\\..\\..\\Apollo11FullColor.jpg");
	}
	protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
	{
		grfx.DrawImage(image, (cx - image.Width)  / 2,
			(cy - image.Height) / 2,
			image.Width, image.Height);
	}
}
